bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/models/form/OptionsConstraint.java

```diff
@@ -63,12 +63,12 @@ default boolean isEnforceEnum() {
     String[] getEnumNames();
 
     /**
-     * Returns a list of RichText to be displayed to the end user.
+     * Returns an array of String or TextContent to be displayed to the end user based on the richText flag.
      * The length of enum and enumNames array must match
      *
      * @return the list of enum names
-     * @since com.adobe.cq.forms.core.components.models.form 5.4.0
+     * @since com.adobe.cq.forms.core.components.models.form 5.5.0
      */
     @JsonProperty("enumNames")
-    TextContent[] getEnumNamesAsTextContent();
+    Object[] getEnumNamesArray();
```

Comment:
Let's keep this array of string, remove the `TextContent` completely from the java code. UI could still have the same flow, let's change the spec too and fix this.